src/JWTVerifier.php

```diff
@@ -177,53 +178,35 @@ public function verifyAndDecode($jwt)
         }
 
         try {
-            $head_decoded = $this->decodeTokenSegment($tks[0]);
             $body_decoded = $this->decodeTokenSegment($tks[1]);
         } catch (\DomainException $e) {
             throw new InvalidTokenException('Malformed token.');
         }
 
-        if (! is_object($head_decoded) || ! is_object($body_decoded)) {
-            throw new InvalidTokenException('Malformed token.');
-        }
-
-        if (empty($head_decoded->alg)) {
-            throw new InvalidTokenException('Token algorithm not found');
-        }
-
-        if (! $this->supportsAlg($head_decoded->alg)) {
-            throw new InvalidTokenException('Token algorithm not supported');
-        }
-
         // Validate the token audience, if present.
         if (! empty($body_decoded->aud)) {
             $audience = is_array($body_decoded->aud) ? $body_decoded->aud : [$body_decoded->aud];
```

Comment:
can we move any claim check below the signature check?

Reply:
It was my impression that we were fixing the big stuff, then addressing the rest in a separate PR. See the WIP branch for the main changes made for this epic.